wgsl/scanner.cc

```diff
@@ -714,16 +716,49 @@ struct Scanner {
         continue;
       }
 
+      // A template list can't contain an assignment or a compound assignment.
+      // There is logic below which clears the stack when reaching one of those.
+      // It looks for a '=' code point.  But we still want to allow
+      // comparison operations inside expressions. So we must pre-emptively
+      // allow operators: == >= <= !=
+
+      // Look for a nested template-list.
       if (lexer.match_identifier()) {
         lexer.skip_whitespace(); // TODO: Skip comments
         if (lexer.match('<')) {
-          LOG("classify_template_args() '<'");
+          LOG("classify_template_args() '<' after ident");
           lt_stack.push_back(StackEntry{state.lt_is_tmpl.count(), expr_depth});
-          state.lt_is_tmpl.push_back(false);  // Default to less-than
+          // Default to less-than (or less-than-equal, or left-shift, or left-shift-equal)
+          state.lt_is_tmpl.push_back(false);
+
+          if (lexer.match('=')) {
+            // Allow "ident<=".
+            // We entered the loop at "ident<=". No template arg can start with '=',
+            // so consider "<=" to be a single token.
+            // Litmus test: "alias z = a<b<=c>;"
+            lt_stack.pop_back();
+          } else if (lexer.match('<')) {
+            // Allow "ident<<".
+            // We entered the loop at "ident<<". No template arg can start with '<',
+            // so consider "<<" to be a single token.
+            // Litmus test: "alias z = a<b<<c>;"
+            state.lt_is_tmpl.push_back(false);
+            lt_stack.pop_back();
+          }
```

Comment:
A few nits:
* `Allow` is an odd phrasing - this is just treating this sequence as an expression, not a template.
* push / pop logic is more complex than I think it needs to be. Suggest:

```c++
        if (lexer.match('<')) {
          LOG("classify_template_args() '<' after ident");
          // Record this '<' in the state.lt_is_tmpl, initially treating this as an expression 
          // (less-than, less-than-equal, left-shift, or left-shift-equal).
          // If this '<' is recorded in lt_stack, and a corresponding '>' is found then
          // this will be transformed into a template start.
          state.lt_is_tmpl.push_back(false);

          if (lexer.match('=')) {  // "ident<=".
            // We entered the loop at "ident<=". No template arg can start with '=',
            // so consider "<=" to be a single token.
            // Litmus test: "alias z = a<b<=c>;"
          } else if (lexer.match('<')) { // "ident<<".
            // We entered the loop at "ident<<". No template arg can start with '<',
            // so consider "<<" to be a single token.
            // Litmus test: "alias z = a<b<<c>;"
            state.lt_is_tmpl.push_back(false);
          } else {
            lt_stack.push_back(StackEntry{state.lt_is_tmpl.count(), expr_depth});
          }
        }
```